tests/TestCase.php

```diff
@@ -78,6 +78,8 @@ protected function getEnvironmentSetUp($app)
         // Set-up admin guard
         $app['config']->set('auth.guards.admin', ['driver' => 'session', 'provider' => 'admins']);
         $app['config']->set('auth.providers.admins', ['driver' => 'eloquent', 'model' => Admin::class]);
+        $app['config']->set('permission.models.permission', \Spatie\Permission\Test\Permission::class);
+        $app['config']->set('permission.models.role', \Spatie\Permission\Test\Role::class);
```

Comment:
Should we add a test that uses the base model directly as well ... to ensure there's no break in that case?

Reply:
I was thinking that, maybe is better remove that ?

Reply:
I guess it depends what we're testing! 😄 
By adding these models and forcing their use in tests, we've excluded directly testing the original model, which most applications will be using. :(

Reply:
So, better i let original model testing and make a new test only for test extended models

Reply:
@drbyte added new test `HasPermissionsWithCustomModelsTest.php` for only test custom models, better readability using collection and eloquent methods